### PR TITLE
TST: stats: fixed misnamed tests

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

part of https://github.com/scipy/scipy/issues/17652
#### What does this implement/fix?

Fixes incorrectly named tests which were previous not ran

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -1197,14 +1197,6 @@ def test_cdf_tail_15110_14753(self):
         assert_allclose(stats.truncnorm(8, np.inf).cdf(8.3),
                         0.9163220907327540)
 
-    def _test_moments_one_range(self, a, b, expected, rtol=1e-7):
```
Comment (j-bowhay) [inline scipy/stats/tests/test_distributions.py:1200]: As far as I can tell this is equivalent to https://github.com/scipy/scipy/blob/5955e906c56fa5f7ec9837f56e44a04a7a4d438f/scipy/stats/tests/test_distributions.py#L1242-L1245

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -4433,7 +4425,7 @@ def test_no_name_arg(self):
         stats.rv_discrete()
 
 
-def TestArgsreduce():
+def test_args_reduce():
```
Comment (j-bowhay) [inline scipy/stats/tests/test_distributions.py:4428]: This test is currently failing. @tupui @mdhaber I'm not really sure what the correct behaviour is, do we still want this test?

**Comment by j-bowhay** [comment]
There is also https://github.com/scipy/scipy/blob/509beb488ae5eabe6f0806af8ded688ac33e31c1/scipy/stats/tests/test_continuous_basic.py#L638-L644 which is not currently called as of #15250. Can it just be removed?

File: scipy/stats/tests/test_distributions.py (hunk 2 above)
Comment (mdhaber) [reply scipy/stats/tests/test_distributions.py:4428]: This looks very old and doesn't seem to belong in `test_distribution`. But please do a little digging as to what it is supposed to do and why it is failing. I don't know offhand.

Comment (j-bowhay) [reply scipy/stats/tests/test_distributions.py:4428]: Looking closer the function it tests is never used and is dead code so can just be removed

**Comment by mdhaber** [comment]
Re: https://github.com/scipy/scipy/pull/17770#issuecomment-1379533068

No, it looks like there was a misunderstanding in gh-15250. I wrote that the distributions could be removed from `fail_loc_scale`, which meant that:
```python3
        if loc_scale_ok:
            check_loc_scale(distfn, arg, m, v, distname)
```
could become:
```python3
        check_loc_scale(distfn, arg, m, v, distname)
```
so that it could be run for all distributions. Instead, it was removed entirely, and I didn't notice the mistake. Please add it back.

File: scipy/stats/tests/test_distributions.py (hunk 2 above)
Comment (j-bowhay) [reply scipy/stats/tests/test_distributions.py]: Turns out it is not dead, I was only searching `.py` files but it is used in the cython wrapper of unu.ran. More investigation required...

File: scipy/stats/tests/test_distributions.py [hunk 3]
```diff
@@ -4433,20 +4425,20 @@ def test_no_name_arg(self):
...
     b, c = argsreduce(2 > 1, a, 2)
-    assert_array_equal(b, a[0])
-    assert_array_equal(c, [2])
+    assert_array_equal(b, a)
+    assert_array_equal(c, [2] * np.size(a))
 
     b, c = argsreduce(a > 0, a, 2)
     assert_array_equal(b, a)
-    assert_array_equal(c, [2] * numpy.size(a))
+    assert_array_equal(c, [2] * np.size(a))
```
Comment (j-bowhay) [inline scipy/stats/tests/test_distributions.py:4441]: There appears to have been a behaviour change to argsreduce in #13611 so I have updated the tests to pass under the assumption that this change in behaviour was desired.

**Comment by j-bowhay** [comment]
Currently one failure
```
_________________________________________________________________________________ test_moments[truncpareto-arg102-True-True-False] __________________________________________________________________________________
[gw4] linux -- Python 3.10.8 /home/jakeb/miniconda3/envs/scipy-dev/bin/python
scipy/stats/tests/test_continuous_basic.py:324: in test_moments
    check_loc_scale(distfn, arg, m, v, distname)
        arg        = (1.8, 5.3)
        distfn     = <scipy.stats._continuous_distns.truncpareto_gen object at 0x7fec57902920>
        distname   = 'truncpareto'
        higher_ok  = True
        is_xfailing = False
        k          = 3.221085799636361
        m          = 1.744069443799767
        normalization_ok = True
        s          = 1.833550308871743
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7fec53cf86d0>
        v          = 0.7076717914498358
scipy/stats/tests/test_continuous_basic.py:643: in check_loc_scale
    mt, vt = distfn.stats(loc=loc, scale=scale, *arg)
        arg        = (1.8, 5.3)
        distfn     = <scipy.stats._continuous_distns.truncpareto_gen object at 0x7fec57902920>
        loc        = array([10., 20.])
        m          = 1.744069443799767
        msg        = 'truncpareto'
        scale      = array([10., 20.])
        v          = 0.7076717914498358
scipy/stats/_distn_infrastructure.py:1109: in stats
    mu = self._munp(1, *goodargs)
        args       = (array(1.8), array(5.3))
        cond       = array([ True,  True])
        default    = array([nan, nan])
        g1         = None
        g2         = None
        goodargs   = [array([1.8, 1.8]), array([5.3, 5.3])]
        kwds       = {'loc': array([10., 20.]), 'scale': array([10., 20.])}
        loc        = array([10., 20.])
        moments    = 'mv'
        mu         = None
        mu2        = None
        output     = []
        scale      = array([10., 20.])
        self       = <scipy.stats._continuous_distns.truncpareto_gen object at 0x7fec57902920>
scipy/stats/_continuous_distns.py:8902: in _munp
    if n == b:
E   ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
        b          = array([1.8, 1.8])
        c          = array([5.3, 5.3])
        n          = 1
        self       = <scipy.stats._continuous_distns.truncpareto_gen object at 0x7fec57902920>
============================================================================================== short test summary info ==============================================================================================
FAILED scipy/stats/tests/test_continuous_basic.py::test_moments[truncpareto-arg102-True-True-False] - ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

**Comment by j-bowhay** [comment]
CI is now green so this is ready for review

**Comment by mdhaber** [comment]
Thanks @j-bowhay! Good to have found that bug.

**Pull request merged**